packages/clerk-js/src/ui/components/ApiKeys/CreateApiKeyForm.tsx

```diff
@@ -0,0 +1,170 @@
+      break;
+    case '90d':
+      future.setDate(future.getDate() + 90);
+      break;
+    case 'custom':
+      future = new Date(customDate as string);
+      break;
+    default:
+      throw new Error('Invalid expiration option');
+  }
+
+  const diffInMs = future.getTime() - now.getTime();
+  const diffInSecs = Math.floor(diffInMs / 1000);
+  return diffInSecs;
+};
+
+const getMinDate = () => {
+  const min = new Date();
+  min.setDate(min.getDate() + 1);
+  return min.toISOString().split('T')[0];
+};
+
+export const CreateApiKeyForm = ({ onCreate, isSubmitting }: CreateApiKeyFormProps) => {
+  const [showAdvanced, setShowAdvanced] = useState(false);
+  const [expiration, setExpiration] = useState<Expiration>('never');
+  const createApiKeyFormId = React.useId();
+  const segmentedControlId = `${createApiKeyFormId}-segmented-control`;
+  const { close: closeCardFn } = useActionContext();
+
+  const nameField = useFormControl('name', '', {
+    type: 'text',
+    label: localizationKeys('formFieldLabel__apiKeyName'),
+    placeholder: localizationKeys('formFieldInputPlaceholder__apiKeyName'),
+    isRequired: true,
+  });
+
+  const descriptionField = useFormControl('description', '', {
+    type: 'text',
+    label: localizationKeys('formFieldLabel__apiKeyDescription'),
+    placeholder: localizationKeys('formFieldInputPlaceholder__apiKeyDescription'),
+    isRequired: false,
+  });
+
+  const expirationDateField = useFormControl('expirationDate', '', {
+    type: 'date',
+    label: localizationKeys('formFieldLabel__apiKeyExpirationDate'),
+    placeholder: localizationKeys('formFieldInputPlaceholder__apiKeyExpirationDate'),
+    isRequired: false,
+  });
+
+  const canSubmit = nameField.value.length > 2;
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    onCreate(
+      {
+        name: nameField.value,
+        description: descriptionField.value || undefined,
+        expiration: getTimeLeftInSeconds(expiration, expirationDateField.value),
+      },
+      closeCardFn,
+    );
+  };
+
+  return (
+    <FormContainer
+      headerTitle={localizationKeys('apiKeys.formTitle')}
+      headerSubtitle={localizationKeys('apiKeys.formHint')}
+    >
+      <Form.Root onSubmit={handleSubmit}>
+        <Form.ControlRow elementId={nameField.id}>
+          <Form.PlainInput {...nameField.props} />
+        </Form.ControlRow>
+        {showAdvanced && (
+          <>
+            <Form.ControlRow elementId={descriptionField.id}>
+              <Form.PlainInput {...descriptionField.props} />
+            </Form.ControlRow>
+            <Col gap={2}>
+              <FormLabel htmlFor={segmentedControlId}>
+                <Text
+                  as='span'
+                  variant='subtitle'
+                  localizationKey={localizationKeys('formFieldLabel__apiKeyExpiration')}
+                />
+              </FormLabel>
+              <SegmentedControl.Root
+                aria-label='Expiration'
```

Comment:
This can be removed as you have aria-labeledby usage.

Reply:
applied!